src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,89 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ *     haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * @todo Should `FormValue`s contain a reference to their
+ *     client object?
+ * 
+ * @constructor
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function(*, *, string, FormValue): *} [makeValid] - 
+ *     returns a valid value no matter what. Will be given the
+ *     new value, the current value, the string 'current' or
+ *     'future', and the `FormValue` instance.
+ * 
+ * @returns {FormValue} formVal
+ * @todo Finish description of FormValue return value (tbd)
+ */
+class FormValue {
+
+  constructor ( name, current, makeValid ) {
+
+    // Set defaults
+    if ( current === undefined ) { current = null; }
+    this._makeValid = makeValid || function ( newVal ) { return newVal; };
```

Comment:
This default implementation differs from the one you've got in your docstring, and in your function invocations in lines 44, 52, 58, and 66. I think it probably needs to change.